diff -ruN caam/arch/arm64/boot/dts/engicam/imx8xq-icore-starterkit.dts caam-b/arch/arm64/boot/dts/engicam/imx8xq-icore-starterkit.dts
--- caam/arch/arm64/boot/dts/engicam/imx8xq-icore-starterkit.dts	2022-08-03 07:54:22.547346956 +0200
+++ caam-b/arch/arm64/boot/dts/engicam/imx8xq-icore-starterkit.dts	2022-08-06 08:02:58.833755667 +0200
@@ -154,6 +154,11 @@
 		};
 	};
 
+	caam_keyblob: caam-keyblob {
+		compatible = "fsl,sec-v4.0-keyblob";
+		status = "okay";
+	};
+
 };
 
 &cm40_intmux {
